src/DotNetty.Codecs.Redis/RedisBulkStringAggregator.cs

```diff
@@ -0,0 +1,219 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace DotNetty.Codecs.Redis
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics.Contracts;
+    using DotNetty.Buffers;
+    using DotNetty.Codecs.Redis.Messages;
+    using DotNetty.Common.Utilities;
+    using DotNetty.Transport.Channels;
+
+    public sealed class RedisBulkStringAggregator : MessageToMessageDecoder<IRedisMessage>
```

Comment:
The origin java implementation has way too much into it for Redis purposes.
I reckon it can be pushed up at some point when other things like proto buf are implemented.